components/candle/scoreService.ts

```diff
@@ -0,0 +1,192 @@
+import {
+    handleEvent,
+    StateMachineLike,
+} from "@peacockproject/statemachine-parser"
+import { getConfig } from "components/configSwizzleManager"
+import { log, LogLevel } from "components/loggingInterop"
+import {
+    CompletedChallengeData,
+    CalculateScoreResult,
+} from "components/types/score"
+import {
+    ClientToServerEvent,
+    CompiledChallengeIngameData,
+    CompiledChallengeRuntimeData,
+} from "components/types/types"
+import { fastClone } from "components/utils"
+
+export class ScoreService {
+    private challengeNameToChallenge: Map<string, CompiledChallengeIngameData> =
+        new Map<string, CompiledChallengeIngameData>()
+    private eventNameToChallengeName: Map<string, string[]> = new Map<
```

Comment:
This really shouldn't be needed, the parser knows how to discard these events.

Reply:
I'm not sure I'm following. What does the parser have to do with this? I need to have a map of all challenges that are affected by an event, so I can replay those in the proper order to calculate the score. Also see `getChallengesForEvent` and then its use in `calculateScore`. For each challenge found to be affected by the event, the state machine is updated accordingly (`handleEvent` - which indeed takes the original event-name, but to get their I need the map).

Reply:
Because the parser is smart enough to deduct when the state machine needs to respond to the event in the first place, so you can just give every event to every state machine and the outcome will be the same, with less code.